framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts

```diff
@@ -63,4 +79,104 @@ export class ChainConnectorPlugin extends BasePlugin<ChainConnectorPluginConfig>
 			await this._sidechainAPIClient.disconnect();
 		}
 	}
+
+	public async getCertificateFromAggregateCommit(
+		aggregateCommit: AggregateCommit,
+	): Promise<Certificate> {
+		const chainConnectorInfo = await getChainConnectorInfo(this._chainConnectorDB);
+
+		const blockHeader = chainConnectorInfo.blockHeaders.find(
+			header => header.height === aggregateCommit.height,
+		)!;
+
+		const certificate = computeCertificateFromBlockHeader(blockHeader);
+		certificate.aggregationBits = blockHeader.aggregateCommit.aggregationBits;
+		certificate.signature = aggregateCommit.certificateSignature;
+
+		return certificate;
+	}
+
+	public async getNextCertificateFromAggregateCommits(
+		lastCertifiedHeight: number,
+		aggregateCommits: AggregateCommit[],
+	): Promise<Certificate | undefined> {
+		const chainConnectorInfo = await getChainConnectorInfo(this._chainConnectorDB);
+
+		const blockHeader = chainConnectorInfo.blockHeaders.find(
+			header => header.height === lastCertifiedHeight,
+		)!;
+
+		const validatorData = chainConnectorInfo.validatorsHashPreimage.find(
+			data => data.validatorsHash === blockHeader.validatorsHash!,
+		)!;
+
+		const blsKeyToBFTWeight: Record<string, bigint> = {};
+
+		for (const validator of validatorData.validators) {
+			blsKeyToBFTWeight[validator.blsKey.toString('hex')] = validator.bftWeight;
+		}
+
+		const bftHeights = await this._mainchainAPIClient.invoke<BFTHeights>('consensus_getBFTHeights');
```

Comment:
Since we are creating a cross chain update for the sidechain we need to call sidechain api client
```suggestion
		const bftHeights = await this._sidechainAPIClient.invoke<BFTHeights>('consensus_getBFTHeights');
```